CMakeLists.txt

```diff
@@ -157,6 +157,11 @@ if( WIN32 AND "${CMAKE_CXX_COMPILER_ID}" MATCHES "Intel" )
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Qlong-double -Qpc80 /DWIN32 /D_WINDOWS /W3 /GR /EHsc -nologo -Od -D_CRT_NONSTDC_NO_WARNINGS -EHsc -Wall -Qdiag-disable:68,111,177,186,161,869,1028,2259,2553,181,239,265,1188 -fp:strict -fp:source")
 endif()
 
+# To handle addresses larger than 2 gigabytes for 32bit targets
+if( WIN32 AND "${CMAKE_EXE_LINKER_FLAGS}" MATCHES "/machine:X86")
```

Comment:
Didn't use the condition suggested [here](https://github.com/KhronosGroup/OpenCL-CTS/issues/1842#issuecomment-1804680022)

> if(MSVC AND ${CLConform_TARGET_ARCH} STREQUAL "x86")

1. Users could use a different compiler (e.g. ICX) with MSVC linker
2. Users could cross-compile 32-bit executables on a 64-bit machine (where `${CLConform_TARGET_ARCH} == "x86_64"`)

Reply:
Thanks @bashbaug for the explanation. It makes sense!